### PR TITLE
brew: bash developer commands and other tweaks

### PR DESCRIPTION
**Pull request by UniqMartin**

### CODE REVIEW

File: Library/Homebrew/cmd/update-bash.sh [hunk 1]
```diff
@@ -1,12 +1,3 @@
-#!/bin/bash
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/update-bash.sh]: Good call on changing to `644`

File: bin/brew [hunk 2]
```diff
@@ -161,7 +168,7 @@ then
   # Hide shellcheck complaint:
   # shellcheck source=/dev/null
   source "$HOMEBREW_BASH_COMMAND"
-  { "$HOMEBREW_COMMAND" "$@"; exit $?; }
+  { "brew-$HOMEBREW_COMMAND" "$@"; exit $?; }
```
Comment (mikemcquaid) [inline bin/brew]: Not sure what I think about this but I guess giving Bash's scoping: :+1:

**Comment by UniqMartin** [comment]
Now also added a commit to fix/extend a test I had previously missed …

**Comment by xu-cheng** [comment]
:+1:

File: bin/brew (hunk 2 above)
Comment (UniqMartin) [inline bin/brew]: Thanks. It shouldn't feel too much off, given our external commands also have to be named `brew-<command>` and our internal Ruby commands are `Homebrew.<command>` (thus also have a prefix). This definitely has the benefit that we won't run into any collisions with shell keywords or shell built-ins, though I hope/assume we don't plan to reimplement major parts of Homebrew in shell. :grinning:

Comment (mikemcquaid) [inline bin/brew]: :cool:. Mind naming this `homebrew-$HOMEBREW_COMMAND` to make the namespacing a bit more similar?

**Comment by mikemcquaid** [comment]
One nit and otherwise :+1: :shipit:

File: bin/brew (hunk 2 above)
Comment (UniqMartin) [inline bin/brew]: > Mind naming this `homebrew-$HOMEBREW_COMMAND` to make the namespacing a bit more similar?

I don't mind at all. In fact that is one of the alternatives I also had in mind but for whatever reason I decided to go with the shorter `brew-` prefix. I'll change it to `homebrew-`.

Comment (mikemcquaid) [inline bin/brew]: Great. Feel free to merge once you've done that :+1:

**Pull request closed without merging**